### PR TITLE
Add gRPC native AOT page

### PR DESCRIPTION
**Pull request by JamesNK**
Add gRPC specific page for native AOT. Will add links to it from the fundamentals AOT page later.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/grpc/native-aot.md](https://github.com/dotnet/AspNetCore.Docs/blob/121040d72cdb98708c785da4c4374407542be30d/aspnetcore/grpc/native-aot.md) | [aspnetcore/grpc/native-aot](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/native-aot?branch=pr-en-us-28899) |

### CODE REVIEW

File: aspnetcore/grpc/native-aot.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# gRPC and native AOT
+
+By [James Newton-King](https://twitter.com/jamesnk)
+
+gRPC supports [.NET native ahead-of-time (AOT)](/dotnet/core/deploying/native-aot/) in .NET 8. Native AOT enables publishing gRPC client and server apps as small, fast native executables.
+
+> [!WARNING]
+> In .NET 8, not all ASP.NET Core features are compatible with native AOT. For more information, see [ASP.NET Core and native AOT compatibility](xref:fundamentals/native-aot#aspnet-core-and-native-aot-compatibility).
+
+## Getting started
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/native-aot.md:19]:
```suggestion
## Get started
```

File: aspnetcore/grpc/native-aot.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+AOT compilation happens when the app is published. Native AOT is enabled with the `PublishAot` option.
+
+01. Add `<PublishAot>true</PublishAot>` to the gRPC client or server app's project file. This will enable native AOT compilation during publish and enable dynamic code usage analysis during build and editing.
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/native-aot.md:23]: Replace numbers with all `1` and build system automatically increments.
Any reason you need 2 digits?

File: aspnetcore/grpc/native-aot.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        <PackageReference Include="Grpc.AspNetCore" Version="2.51.0" />
+        <PackageReference Include="Google.Protobuf" Version="3.22.0" />
+      </ItemGroup>
+    
+    </Project>
+    ```
+
+02. Publish the app for a specific runtime identifier using `dotnet publish -r <RID>`.
+
+The app will be available in the publish directory and will contain all the code needed to run in it.
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/native-aot.md:43]:
```suggestion
The app is available in the publish directory and contains all the code needed to run in it.
```
Per MS style guide, avoid FT.

**Review by Rick-Anderson — APPROVED** [review]

File: aspnetcore/grpc/native-aot.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+* Smaller disk footprint
+* Reduced startup time
+* Reduce memory demand
+
+For more information and examples of the benefits that native AOT provides, see [Benefits of using native AOT with ASP.NET Core](xref:fundamentals/native-aot#benefits-of-using-native-aot-with-aspnet-core).
+
+## Additional resources
+
+* <xref:fundamentals/native-aot>
+* [Native AOT deployment](/dotnet/core/deploying/native-aot/)
```
Comment (mitchdenny) [inline aspnetcore/grpc/native-aot.md:48]: Should we include a link tot he ASP.NET Core native AOT docs as well?

Comment (JamesNK) [reply aspnetcore/grpc/native-aot.md:48]: That's the line above: `<xref:fundamentals/native-aot>`. The title and URL are automatically populated.

File: aspnetcore/grpc/native-aot.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+>
+> For more information, see [ASP.NET Core and native AOT compatibility](xref:fundamentals/native-aot#aspnet-core-and-native-aot-compatibility).
+
+## Get started
+
+AOT compilation happens when the app is published. Native AOT is enabled with the `PublishAot` option.
+
+1. Add `<PublishAot>true</PublishAot>` to the gRPC client or server app's project file. This will enable native AOT compilation during publish and enable dynamic code usage analysis during build and editing.
+
+    [!code-xml[](~/grpc/native-aot/Server.csproj?highlight=5)]
```
Comment (mitchdenny) [inline aspnetcore/grpc/native-aot.md:27]: gRPC deploying to native AOT still needs slim builder right?

File: aspnetcore/grpc/native-aot.md (hunk 4 above)
Comment (mitchdenny) [reply aspnetcore/grpc/native-aot.md:48]: TIL :)

File: aspnetcore/grpc/native-aot.md (hunk 5 above)
Comment (mitchdenny) [reply aspnetcore/grpc/native-aot.md:27]: Ignore ... clarified in team chat. No need for slim builder in this scenario.

**Review by mitchdenny — APPROVED** [review]
LGTM!

File: aspnetcore/grpc/native-aot.md (hunk 5 above)
Comment (JamesNK) [reply aspnetcore/grpc/native-aot.md:25]: No, the normal builder works. The slim builder is an optional optimization.

**Pull request merged**